### ISSUE TITLE
SC2086 info/warning should not be for integer variables

### ISSUE THREAD

**Issue by rogerxxxx** [koalaman/shellcheck#2703]
declare -i debug=0

if [ ${debug_val} -ne 1 ] ; then

^^ SC2086 (info): Double quote to prevent globbing and word splitting.

I do believe globbing and word splitting is not typically used with variables declared as integer.

Simple fix, re-word the warning,

SC2086 (info): Double quote to prevent string globbing and string word splitting.

Anybody will knowingly realize, if their variable is declared as an integer, then this warning is likely not pertinent for their script.  Or further clarify:

SC2086 (info): If this is a string, double quote to prevent string globbing and string word splitting.

#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
your script here

```
